lib/buffer.js

```diff
@@ -107,7 +111,7 @@ function Buffer(arg, encodingOrOffset, length) {
         'If encoding is specified then the first argument must be a string'
       );
     }
-    return Buffer.allocUnsafe(arg);
+    return allocate(arg);
```

Comment:
Yeah, I see that too now. ;)

I don’t see any real disadvantage except that it would basically require duplicating `assertSize`. As said above, if you prefer it that way, you can leave this change here and let later discussion take care of that.